gitlab_jobs.py

```diff
@@ -30,6 +30,19 @@ def get_pipelines(project, args):
             yield pipeline
 
 
+def get_jobs(pipeline, scope, all):
+    page = 0
+    while True:
+        page += 1
+
+        jobs = pipeline.jobs.list(
+                scope=scope, all=all, page=page, per_page=100)
+        if len(jobs) == 0:
+            raise StopIteration()
```

Comment:
I would use a `break` here.

Reply:
Actually `raise StopIteration()` is a bug on newer Python versions (such as 3.7, which makes it raise RuntimeError: generator raised StopIteration), and you should really get into the habit of using `return` instead of it.

Reply:
My experiments indicate that yes, all returns everything, so this PR actually introduced a bug that causes repeated jobs :(

Reply:
Oops these comments belong in the other thread!